src/main/scala/com/yotpo/metorikku/code/steps/SelectiveMerge.scala

```diff
@@ -0,0 +1,70 @@
+package com.yotpo.metorikku.code.steps
+
+import com.yotpo.metorikku.exceptions.MetorikkuException
+import org.apache.log4j.{LogManager, Logger}
+import org.apache.spark.sql.{Column, DataFrame}
+import org.apache.spark.sql.functions._
+
+object SelectiveMerge {
+  val message = "You need to send 3 parameters with the names of the dataframes to merge and the key(s) to merge on" +
+    "(merged df1 into df2 favouring values from df2): df1, df2, Seq[String]"
+
+  private val log: Logger = LogManager.getLogger(this.getClass)
+
+  private val colRenamePrefix = "df2_"
+
+  def run(ss: org.apache.spark.sql.SparkSession, metricName: String, dataFrameName: String, params: Option[Map[String, String]]): Unit = {
+    params match {
+      case Some(parameters) => {
+        val df1 = ss.table(parameters.get("df1").get)
+        val df2 = ss.table(parameters.get("df2").get)
+        val joinKeys = parameters.get("joinKeys").get.split(" ").toSeq
+
+        if (df2.isEmpty) {
```

Comment:
I think we want to log error if df1 is empty

Reply:
done